### PR TITLE
[Merged by Bors] - Operator Discovery

### PR DESCRIPTION
**Pull request by DmitryDodzin**
contains #768.

Auto discover the operator in cluster and portforward into it

### CODE REVIEW

File: mirrord-schema.json [hunk 1]
```diff
@@ -382,6 +392,26 @@
...
+      "type": "object",
+      "properties": {
+        "addr": {
+          "description": "Direct Connect to operator addr",
+          "type": [
+            "string",
+            "null"
+          ]
+        },
+        "enabeld": {
```
Comment (aviramha) [inline mirrord-schema.json:406]: typo

File: mirrord-schema.json [hunk 2]
```diff
@@ -382,6 +392,26 @@
...
       "additionalProperties": false
     },
+    "OperatorFileConfig": {
+      "description": "Configuration for the mirrord-agent pod that is spawned in the Kubernetes cluster.",
+      "type": "object",
+      "properties": {
+        "addr": {
+          "description": "Direct Connect to operator addr",
+          "type": [
+            "string",
```
Comment (aviramha) [inline mirrord-schema.json:402]: how does that work?

File: mirrord/cli/src/main.rs [hunk 3]
```diff
@@ -108,6 +109,20 @@ fn add_to_preload(path: &str) -> Result<()> {
 #[tokio::main(flavor = "current_thread")]
 async fn create_agent(progress: &TaskProgress) -> Result<()> {
     let config = LayerConfig::from_env()?;
+
+    if config.operator.enabeld {
```
Comment (aviramha) [inline mirrord/cli/src/main.rs:113]: `enabled`

File: mirrord/layer/src/connection.rs [hunk 4]
```diff
@@ -48,17 +49,22 @@ pub(crate) async fn connect(
...
-            .create_connection(stream)
+            .unwrap_or_else(|err| handle_error(err, config))
+    } else if config.operator.enabeld && let Some((operator_api, operator_ref)) =
+        OperatorApiDiscover::discover_operator(config, &progress).await
+    {
+        operator_api
+            .create_connection(operator_ref)
             .await
             .unwrap_or_else(|err| handle_error(err, config))
+    } else if let Some(addr) = &config.operator.addr {
```
Comment (aviramha) [inline mirrord/layer/src/connection.rs:63]: when will you be able to connect to operator like this? is this for testing purposes or user use case?

**Review by aviramha — CHANGES_REQUESTED** [review]
Almost there!

File: mirrord-schema.json (hunk 2 above)
Comment (DmitryDodzin) [reply mirrord-schema.json]: removed it

**Review by aviramha — APPROVED** [review]
LGTM!

**Pull request closed without merging**